src/cascade/input_data/db/csmr.py

```diff
@@ -0,0 +1,136 @@
+"""Upload csmr data to t3 table so EpiViz can access it for plotting."""
+
+import logging
+
+try:
+    from db_queries import get_outputs
+except ImportError:
+
+    class DummyGetOutputs:
+        def __getattr__(self, name):
+            raise ImportError(f"Required package db_queries not found")
+
+    get_outputs = DummyGetOutputs()
+
+
+from cascade.core.db import cursor
+
+
+CODELOG = logging.getLogger(__name__)
+
+
+def _csmr_in_t3(execution_context):
+    """Checks if data for the current model_version_id already exists in tier 3.
+    """
+
+    model_version_id = execution_context.parameters.model_version_id
+
+    query = """
+    SELECT exists(
+             SELECT * FROM epi.t3_model_version_csmr
+             WHERE model_version_id = %(model_version_id)s
+    )
+    """
+    with cursor(execution_context) as c:
+        c.execute(query, args={"model_version_id": model_version_id})
+        exists = c.fetchone()[0]
+
+    return exists == 1
+
+
+def _get_csmr_data(execution_context):
+
+    model_version_id = execution_context.parameters.model_version_id
+    add_csmr_cause = execution_context.parameters.add_csmr_cause
+    parent_loc = execution_context.parameters.drill
+
+    csmr_measure_id_deaths = 1  # measure type = deaths (1)
+    csmr_metric_id_rate = 3  # metric type = per capita rate
+    csmr_gbd_round_id = 5
+
+    csmr_keep_cols = ["year_id", "location_id", "sex_id", "age_group_id",
+                      "val", "lower", "upper"]
+
+    csmr = get_outputs(
+        topic="cause",
+        cause_id=add_csmr_cause,
+        location_id=parent_loc,
+        metric_id=csmr_metric_id_rate,
+        year_id="all",
+        age_group_id="most_detailed",
+        measure_id=csmr_measure_id_deaths,
+        sex_id="all",
+        gbd_round_id=csmr_gbd_round_id
+        )[csmr_keep_cols]
+
+    csmr["model_version_id"] = model_version_id
+
+    csmr_columns = ["model_version_id"] + csmr_keep_cols
+    csmr = csmr[csmr_columns]
+
+    csmr.rename(columns={"val": "mean"}, inplace=True)
```

Comment:
Let's avoid inplace operations for the sake of consistency. In general they're no cheaper.